OpenRA.Mods.Common/Pathfinder/HierarchicalPathFinder.cs

```diff
@@ -1112,13 +1135,13 @@ List<GraphConnection> AbstractEdge(CPos abstractCell)
 				var maybeAbstractCell = AbstractCellForLocalCellNoAccessibleCheck(cell);
 				if (maybeAbstractCell == null)
 				{
-					// If the source cell in unreachable, use one of the adjacent reachable cells instead.
+					// If the source cell is unreachable, use one of the adjacent reachable cells instead.
 					if (sources != null && sources.Contains(cell))
 					{
 						foreach (var dir in CVec.Directions)
 						{
 							var adjacentSource = cell + dir;
-							if (!world.Map.Contains(adjacentSource))
+							if (!world.Map.Contains(adjacentSource) || unpathableNodes.Contains(adjacentSource))
```

Comment:
this can return `System.NullReferenceException`

Reply:
If we pass non-null `sources`, we also provide non-null `unpathableNodes`.

Reply:
```suggestion
							if (!world.Map.Contains(adjacentSource) || (unpathableNodes != null && unpathableNodes.Contains(adjacentSource)))
```

Reply:
> If we pass non-null `sources`, we also provide non-null `unpathableNodes`.

Well the game just crashed for me

Reply:
And this suggested fix, repaired my replay

Reply:
Ah you are correct. Fixed.